rpc/client/httpclient.go

```diff
@@ -3,25 +3,30 @@ package client
 import (
 	"context"
 	"sync"
+	"time"
 
 	"github.com/pkg/errors"
 
 	amino "github.com/tendermint/go-amino"
+
 	cmn "github.com/tendermint/tendermint/libs/common"
-	tmpubsub "github.com/tendermint/tendermint/libs/pubsub"
 	ctypes "github.com/tendermint/tendermint/rpc/core/types"
 	rpcclient "github.com/tendermint/tendermint/rpc/lib/client"
 	"github.com/tendermint/tendermint/types"
 )
 
 /*
-HTTP is a Client implementation that communicates
-with a tendermint node over json rpc and websockets.
+HTTP is a Client implementation that communicates with a tendermint node over
+json rpc and websockets.
+
+This is the main implementation you probably want to use in production code.
+There are other implementations when calling the tendermint node in-process
+(Local), or when you want to mock out the server for test code (mock).
 
-This is the main implementation you probably want to use in
-production code.  There are other implementations when calling
-the tendermint node in-process (local), or when you want to mock
-out the server for test code (mock).
+You can subscribe for any event published by Tendermint using Subscribe method.
+Note delivery is best-effort. If you don't read events fast enough or network
+is slow, Tendermint might cancel the subscription. The client will attempt to
+resubscribe (you don't need to do anything).
```

Comment:
Do we need to build in the resubscription right away? I guess before it wasn't supposed to be possible to loose messages, because it blocked until you read it. Now the event stream could fail if you're too slow, even if the WS is still active, so we could try to reconnect server side, but what if you meanwhile missed some events, or otherwise can't reconnect? We'll need to expose that to the client so they can figure out what they missed, or else implement much more complex logic server side to do it for them ...

Reply:
> I guess before it wasn't supposed to be possible to loose messages, because it blocked until you read it.

If a client does not read messages, server's output queue will become full and `TryWriteRPCResponse` will return false.

> Do we need to build in the resubscription right away?

It's not idea, but a good start. We don't have to spend too much time on it, but I think having resubscription is beneficial for any client.